scene/main/custom_resource_loader.h

```diff
@@ -0,0 +1,23 @@
+#ifndef CUSTOM_RESOURCE_LOADER_H
+#define CUSTOM_RESOURCE_LOADER_H
+
+#include "scene/main/node.h"
+
+class CustomResourceLoader : public Node, public ResourceFormatLoader {
```

Comment:
Multiple inheritance fucks up the godot cast system on some platforms due to pointer adjustment, so it cant be used. Maybe instead, the system should be modified for ResourceFormatLoader and ResourceFormatSaver to inherit Reference

Reply:
more in detail, using multiple inheritance breaks godot type system when disabling rtti

Reply:
I can implement with composition then. The thing is, I'd really like to have this feature in 3.1 because it makes my plugin really hard to use